src/createNavigationContainer.js

```diff
@@ -160,16 +175,7 @@ export default function createNavigationContainer<T: *>(
       }
       const nav = Component.router.getStateForAction(action, state.nav);
```

Comment:
I am suggesting to defer this change to a separate issue as there are few concerns raised, including the synchronous return value of `dispatch` that is used in e.g. `navigate` as a return value.

Not sure how to handle such case. Rather than that, this should be good to go.